### PR TITLE
Melzer/iscsi

### PR DESCRIPTION
**Pull request by melzer-b1**
Check if iscsi-formula should be installed on node. It is only needed on nodes which use sbd and on the iscsi server.

### CODE REVIEW

**Review by yeoldegrove — CHANGES_REQUESTED** [review]
I find it quite tricky to remove iscsi from `os_setup` as e.g. `kernel-default` can only be installed here (`cluster_node` state will not trigger a reboot).
Nevertheless, does it make sense to remove `iscsi-formula` completely from `os_setup`?
@arbulu89 What do you think?

**Comment by arbulu89** [comment]
Hey @melzer-b1 ,
I don't think we use the `iscsi-formula` in the iscsi initiator in the nodes.
Have you tried the current changes without that part?

Anyway, as @yeoldegrove commented, maybe it makes more sense to remove this from `os_setup` and put it in the `iscsi_srv` folder, as the role based filtering would be already done there

**Comment by melzer-b1** [comment]
I've tested the changes without this part and removed it.

**Comment by yeoldegrove** [comment]
@melzer-b1 As @arbulu89 pointed out... Is `iscsi-formula` really needed on the the iscsi initiators (clients)?
Please double check.

File: salt/os_setup/packages.sls [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# check if iscsi-formula should be installed
```
Comment (arbulu89) [inline salt/os_setup/packages.sls:1]: As commented previously, to make the code better and use the roles properly, we should just move the `iscsi-formula` installation from `os_setup/packages` to `iscsi_srv` folder (we could create there a `packages.sls` file if needed). This way, the role filtering would be already done and we wouldn't need any `if` statement, as `iscsi_srv` is only executed in iscsi servers which, and the user already has selected to use it

**Comment by melzer-b1** [comment]
@arbulu89 
I removed the first part of the condition. So iscsi-formula is only  installed on node with role iscsi_srv. It works.

**Comment by arbulu89** [comment]
> @arbulu89
> I removed the first part of the condition. So iscsi-formula is only installed on node with role iscsi_srv. It works.

Hey @melzer-b1 ,
I'm sure it works, but I still think the solution proposed in the next comment is the way to go:https://github.com/SUSE/ha-sap-terraform-deployments/pull/731#discussion_r677156854
Unless it has some major contraindication, that I cannot think of

**Comment by melzer-b1** [comment]
@arbulu89 the iscsi-formula is now installed from within iscsi_srv. It is removed from os_setup.
We had to solve a chicken egg problem here. Because the installed package iscsi-formula requires further salt run we
introduced role:iscsi_srv in deployment to be run after predeployment. We also had to modify the provision.sh script.

**Comment by arbulu89** [comment]
> @arbulu89 the iscsi-formula is now installed from within iscsi_srv. It is removed from os_setup.
> We had to solve a chicken egg problem here. Because the installed package iscsi-formula requires further salt run we
> introduced role:iscsi_srv in deployment to be run after predeployment. We also had to modify the provision.sh script.

Ok...
Now I understand why we had the `iscsi-formula` installation there. I remember that we had this exact problem.
Knowing this, I would prefer to go back to the https://github.com/SUSE/ha-sap-terraform-deployments/pull/731/commits/fb910b628640adfeba865a8af7378978e1332d3b
The code was simpler.
Sorry, for all the inconveniences...
Once you have it, we will merge it, as this commit was tested. Just remove the last code.
Thanks @melzer-b1

**Comment by melzer-b1** [comment]
Rolled git branch back to first solution.

**Review by arbulu89 — APPROVED** [review]
Thanks @melzer-b1

**Pull request merged**